silkworm/execution/local_state.cpp

```diff
@@ -27,7 +28,12 @@ using namespace db::state;
 using namespace datastore;
 
 std::optional<Account> LocalState::read_account(const evmc::address& address) const noexcept {
-    AccountsDomainGetLatestQuery query{tx_, data_store_.state_db().accounts_domain()};
+    AccountsDomainGetLatestQuery query{
+        db::state::kDomainNameAccounts,
+        data_store_.chaindata,
+        tx_,
+        data_store_.state_repository,
+    };
```

Comment:
AccountsDomainGetLatestQuery only deals with the `db::state::kDomainNameAccounts` domain, so ideally the declaration would be:

```suggestion
    AccountsDomainGetLatestQuery query{
        data_store_.chaindata,
        tx_,
        data_store_.state_repository,
    };
```

Reply:
good idea, I'll try to hide it